ddl/backfilling.go

```diff
@@ -279,10 +304,11 @@ func (w *worker) waitTaskResults(workers []*backfillWorker, taskCnt int, totalAd
 			*totalAddedCount += int64(result.addedCount)
 			addedCount += int64(result.addedCount)
 			nextHandle = result.nextHandle
+			warnings, warningsCount = mergeWarningsAndWarningsCount(result.warnings, warnings, result.warningsCount, warningsCount)
```

Comment:
We can make warnings as `map[errors.ErrorID]*terror.Error` type, then we merge warnings and warnings-count more easily.

Then we can do [update warnings type](https://github.com/pingcap/tidb/pull/20012/files?file-filters%5B%5D=.go#diff-0ef842ac66131b78e2b1b259c81fc5f4R1202-R1205) before line311.